src/main/kotlin/app/revanced/patches/twitter/misc/fxtwitter/FxTwitterPatch.kt

```diff
@@ -0,0 +1,134 @@
+                            overrideString = string.replace("x.com", shareLinkBase.toString())
+                        }
+
+                        this.mutableMethod.replaceInstruction(
+                            index,
+                            """
+                                const-string v$overrideRegister, "$overrideString"
+                            """
+                        )
+                    }
+                }
+            }
+        } ?: throw PatchException("FxTwitterPatchFingerprint not found")
+
+
+        FxTwitterPatchFingerprintResource.result?.apply {
+            val instructions = this.mutableMethod.getInstructions()
+
+            // Save user nickname to v9
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (
+                        methodRef.returnType != "Ljava/lang/String;") continue;
+
+                    val sourceRegister = (instructions[index + 1] as OneRegisterInstruction).registerA
+
+                    this.mutableMethod.addInstruction(
+                        index + 2,
+                        """
+                            move-object v9, v$sourceRegister
+                        """
+                    )
+
+                    break;
+                }
+            }
+
+            // Save tweet id to v7, v8
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (methodRef.definingClass != "Ljava/lang/Long;") continue;
+
+                    val sourceRegister = (instructions[index + 1] as OneRegisterInstruction).registerA
+
+                    this.mutableMethod.addInstruction(
+                        index + 2,
+                        """
+                            move-wide v7, v$sourceRegister
+                        """
+                    )
+                    break;
+                }
+            }
+
+            // Call the method patched by FxTwitterPatchFingerprint
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (methodRef.definingClass != "Landroid/content/res/Resources;") continue;
+
+                    this.mutableMethod.replaceInstruction(
+                        index,
+                        """
+                            invoke-static { v7, v8, v9 }, ${twitterGetShareLinkMethod.method}
+                        """
```

Comment:
Hardcoding the register can fail, you should get the register from the instruction and reuse it here.